rest_auth/registration/serializers.py

```diff
@@ -186,3 +182,44 @@ def save(self, request):
 
 class VerifyEmailSerializer(serializers.Serializer):
     key = serializers.CharField()
+
+
+# Import is needed only if we are using social login, in which
+# case the allauth.socialaccount will be declared
+if 'allauth.socialaccount' in settings.INSTALLED_APPS:
+    from allauth.socialaccount.helpers import complete_social_login
+    from allauth.socialaccount.models import SocialAccount
+    from allauth.socialaccount.providers.base import AuthProcess
+
+    class SocialAccountSerializer(serializers.ModelSerializer):
+        """
+        serialize allauth SocialAccounts for use with a REST API
+        """
+        class Meta:
+            model = SocialAccount
+            fields = (
+                'id',
+                'provider',
+                'uid',
+                'last_login',
+                'date_joined',
+                'extra_data',
```

Comment:
@aleksihakli Should we return extra data here? Depending on provider it might contain some sensitive info that we do not want to expose to frontend my default. For example here's what facebook social account info would display:

```
"extra_data": "{'verified': True, 'locale': 'en_US', 'first_name': 'Maxim', 'id': '{my_facebook_user_id}', 'name': 'Maxim Kukhtenkov', 'link': 'https://www.facebook.com/app_scoped_user_id/{my_facebook_user_id}/', 'updated_time': '2017-08-27T01:17:37+0000', 'email': 'myfacebookemail@example.com', 'last_name': 'Kukhtenkov', 'gender': 'male', 'timezone': -5}"
```

Also notice how it's not parsed as JSON and returned as plain string

Reply:
You're right, it's probably better to be a bit stricter. I removed the `extra_data` attribute.